src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1457,7 +1457,7 @@ private MethodSymbol FindEntryPoint(CancellationToken cancellationToken, out Imm
                 var viableEntryPoints = ArrayBuilder<MethodSymbol>.GetInstance();
                 foreach (var candidate in entryPointCandidates)
                 {
-                    if (!candidate.HasEntryPointSignature(this))
+                    if (!candidate.HasEntryPointSignature(this, diagnostics))
```

Comment:
> Tests for all of the permutations of main method declarations are going in the next PR.

I am not asking for all of the permutations. I am asking to add a test relevant to the particular change you are making on this line. I believe you are introducing a bug here, so a test is warranted.